driver-core/src/main/com/mongodb/client/model/TimeSeriesOptions.java

```diff
@@ -104,12 +109,102 @@ public TimeSeriesOptions granularity(@Nullable final TimeSeriesGranularity granu
         return this;
     }
 
+    /**
+     * Returns the maximum time span between measurements in a bucket.
+     *
+     * @param timeUnit the time unit.
+     * @return time span between measurements, or {@code null} if not set.
+     * @since 4.10
+     * @mongodb.server.release 6.3
+     * @mongodb.driver.manual core/timeseries-collections/ Time-series collections
+     * @see #bucketMaxSpan(Long, TimeUnit)
+     */
+    @Nullable
+    public Long getBucketMaxSpan(final TimeUnit timeUnit) {
+        if (bucketMaxSpanSeconds == null) {
+            return null;
+        }
+        return timeUnit.convert(bucketMaxSpanSeconds, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Sets the maximum time span between measurements in a bucket.
+     * <p>
+     * The value of {@code bucketMaxSpanSeconds} must be the same as {@code bucketRoundingSeconds}.
+     * If you set the {@code bucketMaxSpanSeconds}, parameter, you can't set the granularity parameter.
```

Comment:
Regarding 1, I agree. I will add it accordingly

Regarding 2, I understand convenience for having the `granularity` option erase `bucketMaxSpan` and `bucketRounding` to avoid potential confusion. However, I'm concerned that this approach might lead to unintended side effects for users. There is a possibility that users might accidentally specify `bucketMaxSpan` and then `granularity `when they only intended to set `bucketMaxSpan`. In such cases, if `granularity` silently erases `bucketMaxSpan`, the user's request would be successful on the server side, and they might spend a significant amount of time debugging or remain unaware of the misconfiguration/bug. To prevent this, i would prefer to follow your suggestion of throwing an IllegalStateException when a conflict is detected. This approach would provide immediate feedback to users, clearly indicating the misconfiguration and conflict.

Let me know if you have any further suggestions